init/pkg/create-aerospike-conf.go

```diff
@@ -58,117 +62,107 @@ func (initp *InitParams) createAerospikeConf() error {
+		if initp.networkInfo.HeartBeatPort != 0 {
 			strdataSplit := strings.Split(confString, "heartbeat {")
 			confString = fmt.Sprintf("%sheartbeat {\n        address %s%s",
-				strdataSplit[0], os.Getenv("MY_POD_IP"), strdataSplit[1])
+				strdataSplit[0], initp.networkInfo.podIP, strdataSplit[1])
 		}
 
-		if initp.initTemplateInput.HeartBeatTLSPort != 0 {
+		if initp.networkInfo.HeartBeatTLSPort != 0 {
 			strdataSplit := strings.Split(confString, "heartbeat {")
 			confString = fmt.Sprintf("%sheartbeat {\n        tls-address %s%s",
-				strdataSplit[0], os.Getenv("MY_POD_IP"), strdataSplit[1])
+				strdataSplit[0], initp.networkInfo.podIP, strdataSplit[1])
 		}
 
-		if initp.initTemplateInput.FabricPort != 0 {
+		if initp.networkInfo.FabricPort != 0 {
 			strdataSplit := strings.Split(confString, "fabric {")
 			confString = fmt.Sprintf("%sfabric {\n        address %s%s",
-				strdataSplit[0], os.Getenv("MY_POD_IP"), strdataSplit[1])
+				strdataSplit[0], initp.networkInfo.podIP, strdataSplit[1])
 		}
 
-		if initp.initTemplateInput.FabricTLSPort != 0 {
+		if initp.networkInfo.FabricTLSPort != 0 {
 			strdataSplit := strings.Split(confString, "fabric {")
 			confString = fmt.Sprintf("%sfabric {\n        tls-address %s%s",
-				strdataSplit[0], os.Getenv("MY_POD_IP"), strdataSplit[1])
+				strdataSplit[0], initp.networkInfo.podIP, strdataSplit[1])
 		}
 	}
 
-	if err = os.WriteFile(CFG, []byte(confString), 0644); err != nil { //nolint:gocritic,gosec // file permission
+	if err = os.WriteFile(cfg, []byte(confString), 0644); err != nil { //nolint:gocritic,gosec // file permission
 		return err
 	}
 
+	logrus.Info("aerospike.template.conf = ", confString)
+
 	return nil
 }
 
-func (initp *InitParams) substituteEndpoint(addressType, confString string) string {
+func (initp *InitParams) substituteEndpoint(networkType asdbv1beta1.AerospikeNetworkType,
+	addressType, confString string) string {
 	var (
 		accessAddress string
 		accessPort    int32
-		networkType   asdbv1beta1.AerospikeNetworkType
 		podPort       int32
 		mappedPort    int32
 	)
 
-	if addressType == Access || addressType == AlternateAccess {
-		if addressType == Access {
-			networkType = initp.initTemplateInput.NetworkPolicy.AccessType
-		} else {
-			networkType = initp.initTemplateInput.NetworkPolicy.AlternateAccessType
-		}
-
-		podPort = initp.initTemplateInput.PodPort
-		mappedPort = initp.mappedPort
+	if addressType == access || addressType == alternateAccess {
+		podPort = initp.networkInfo.PodPort
+		mappedPort = initp.networkInfo.mappedPort
 	}
 
-	if addressType == TLSAccess || addressType == TLSAlternateAccess {
-		if addressType == TLSAccess {
-			networkType = initp.initTemplateInput.NetworkPolicy.TLSAccessType
-		} else {
-			networkType = initp.initTemplateInput.NetworkPolicy.TLSAlternateAccessType
-		}
-
-		podPort = initp.initTemplateInput.PodTLSPort
-		mappedPort = initp.mappedTLSPort
+	if addressType == tlsAccess || addressType == tlsAlternateAccess {
```

Comment:
See if we can use this `if os.Getenv("MY_POD_TLS_ENABLED") == "true"` instead just like at other places

Reply:
why to do a os call, if we can simply use this if.

Reply:
Both ways are fine, just added comment to keep this TLS differentiator flow in sync. At one place we use that MY_POD_TLS_ENABLED var